File: Typescript/demos/browser/features/step_definitions/browserTestSteps.ts
import { Given, Then } from "@cucumber/cucumber";
import { testEvolve } from "@testevolve/testevolve-spark";
import {homepage} from "../support/pages";

Given(/^I visit a page$/, async () => {
    await testEvolve.browser.goto("https://www.testevolve.com/element-capture");
});

Given(/^I goto a site$/, async () => {
    await testEvolve.browser.goto("https://www.testevolve.com/element-capture");
});

Given(/^I visit a page - (.*)$/, async (page) => {
    await testEvolve.browser.goto(page);
});

Given('I visit a webpage', async (table) => {
    await testEvolve.browser.goto(table.hashes()[0].page);
});

Given('I visit a website', async (page) => {
    await testEvolve.browser.goto(page);
});

Given(/^I go to a site$/, async () => {
    await testEvolve.browser.goto("https://www.testevolve.com/element-capture");
});

Given(/^I go to a site - (.*)$/, async (page) => {
    await testEvolve.browser.goto(page);
});

Given('I go to a web-page', async (table) => {
    await testEvolve.browser.goto(table.hashes()[0].page);
});

Given('I go to a web-site', async (page) => {
    await testEvolve.browser.goto(page);
});

Then(/^I interact with page elements via the Spark API$/, async () => {
    await homepage.interactWithPageObjectsAPI();
    await homepage.assertObjectChanges();
    await testEvolve.audit("audit");
    await testEvolve.visualCheck("visual");
});

File: Ruby Free/demos/browser/features/step_definitions/browser_test_steps.rb
Then(/^I interact with page elements$/) do
  homepage.interact_with_page_objects
  homepage.assert_object_changes
  TestEvolve.audit 'homepage'
  TestEvolve.visual_check 'homepage'
end

Given(/^I load a page$/) do
  TE.browser.goto('https://testevolve.com/element-capture')
end

Given(/^I load a page - (.*)$/) do |page|
  TE.browser.goto page
end

Given(/^I load a webpage$/) do |table|
  table.hashes.each do |value|
    TE.browser.goto value[:page]
  end
end

Given(/^I go to a page$/) do
  TE.browser.goto('https://testevolve.com/element-capture')
end

Given(/^I go to a page - (.*)$/) do |page|
  TE.browser.goto page
end

Given(/^I go to a webpage$/) do |table|
  table.hashes.each do |value|
    TE.browser.goto value[:page]
  end
end

Given(/^I load a website$/) do |text|
  TE.browser.goto text
end

Given(/^I go to a website$/) do |text|
  TE.browser.goto text
end